### PR TITLE
chore: Introduce basic deadlock detector for tests

### PR DESCRIPTION
**Pull request by romange**

### CODE REVIEW

File: src/server/test_utils.cc [hunk 1]
```diff
@@ -201,6 +211,20 @@ void BaseFamilyTest::ResetService() {
...
+void BaseFamilyTest::ShutdownService() {
+  DCHECK(service_);
+
+  service_->Shutdown();
+  service_.reset();
+  delete shard_set;
+  shard_set = nullptr;
+
+  watchdog_done_.Notify();
+  watchdog_fiber_.Join();
```
Comment (adiholden) [inline src/server/test_utils.cc:223]: you are calling join twice if ResetService is called from a test
once from ResetService and another time from TearDown

Comment (romange) [reply src/server/test_utils.cc:224]: fixed

**Review by adiholden — APPROVED** [review]

File: src/server/test_utils.cc [hunk 2]
```diff
@@ -199,6 +202,28 @@ void BaseFamilyTest::ResetService() {
 
   const TestInfo* const test_info = UnitTest::GetInstance()->current_test_info();
   LOG(INFO) << "Starting " << test_info->name();
+
+  watchdog_fiber_ = pp_->GetNextProactor()->LaunchFiber([this] {
+    if (!watchdog_done_.WaitFor(30s)) {
```
Comment (royjacobson) [inline src/server/test_utils.cc:207]: I'd increase this a bit, maybe 100-120s?

**Review by royjacobson — DISMISSED** [review]

File: src/server/test_utils.cc (hunk 2 above)
Comment (romange) [reply src/server/test_utils.cc]: done

**Review by royjacobson — APPROVED** [review]

**Pull request merged**